GitExtUtils/GitArgumentBuilder.cs

```diff
@@ -171,22 +98,40 @@ public void Add(GitConfigItem configItem)
             }
         }
 
-        /// <inheritdoc />
-        public override string ToString()
+        /// <summary>
+        /// Gets the length of the expected command.
+        /// </summary>
+        /// <returns>THe length of the expected command.</returns>
+        public override int GetLength()
         {
-            var arguments = base.ToString();
             var gitArgsLength = _gitArgs.Length;
             if (gitArgsLength == 0)
             {
                 gitArgsLength = -1; // prevent extra capacity when length is 0
             }
 
             // 7 = "-c " + "=" + " " + 2 (for possible quotes of value)
-            var capacity = _configItems.Sum(i => i.Key.Length + i.Value.Length + 7) + _command.Length + 1 + arguments.Length;
-            capacity += gitArgsLength + 1;
+            var length = _configItems.Sum(i => i.Key.Length + i.Value.Length + 7) +
```

Comment:
`Sum` here will allocate. `foreach` over `List<>` will not.

Reply:
Ah yes, I keep overlooking Linq often allocates
Thank you 👍

Reply:
@drewnoakes  Pardon my ignorance, what do you mean `Sum` allocates? Doesn't a a foreach also allocate an iterator.

Reply:
@kbilsted for `IEnumerable<>` yes there is an allocation, but `List<>` has a struct enumerator (which is not boxed) so avoids the allocation.